src/Shared/UrlDecoder/UrlDecoder.cs

```diff
@@ -345,5 +344,296 @@ private static bool SkipUnescape(int value1, int value2, bool isFormEncoding)
 
             return false;
         }
+
+        /// <summary>
+        /// Unescape a URL path
+        /// </summary>
+        /// <param name="source">The char span represents a UTF8 encoded, escaped url path.</param>
+        /// <param name="destination">The char span where unescaped url path is copied to.</param>
+        /// <returns>The length of the char sequence of the unescaped url path.</returns>
+        /// <remarks>
+        /// Form Encoding is not supported compared to the <see cref="DecodeRequestLine(ReadOnlySpan{byte}, Span{byte}, bool)" />
+        /// for performance gains, as current use-cases does not require it.
+        /// </remarks>
+        public static int DecodeRequestLine(ReadOnlySpan<char> source, Span<char> destination)
+        {
+            if (destination.Length < source.Length)
```

Comment:
`Span<T>.CopyTo` already performs this check and throws the correct exception type and parameter name ("destination"). No need to duplicate.

Reply:
ack